### PR TITLE
Implement Audio

### PR DESCRIPTION
**Pull request by raffclar**
#273

### CODE REVIEW

**Comment by bwrsandman** [comment]
Fixes done in #621

**Comment by bwrsandman** [comment]
The cross compile issues are due to https://github.com/kcat/openal-soft/issues/871

File: .clang-format-ignore [hunk 1]
```diff
@@ -6,3 +6,5 @@ src/Common/stb_image_write.h
 test/mobile_wall_hug/json.hpp
 externals/imgui_user/imgui_impl_sdl.h
 externals/imgui_user/imgui_impl_sdl.cpp
+externals/dr_libs/dr_mp3.h
```
Comment (raffclar) [inline .clang-format-ignore:9]: Why not use vcpkg?

Comment (bwrsandman) [reply .clang-format-ignore:9]: Because the package is not common on linux platforms and ti would break the system-deps

**Comment by bwrsandman** [comment]
Putting to draft until the git module change commit is remove (i.e. vcpkg's openal-soft pr gets merged).

**Review by bwrsandman — APPROVED** [review]

**Pull request merged**